File: src/test/java/nl/mpcjanssen/simpletask/sort/ContextComparatorTest.java
package nl.mpcjanssen.simpletask.sort;

import junit.framework.TestCase;
import nl.mpcjanssen.simpletask.task.Task;

import java.util.ArrayList;
import java.util.Collections;
import java.util.List;

/**
 * Created by Mark on 2015-07-16.
 */
public class ContextComparatorTest extends TestCase {

    public void testCompare() throws Exception {
        List<Task> tasks = new ArrayList<>();
        tasks.add(new Task("Test @b"));
        tasks.add(new Task("Test @a"));
        tasks.add(new Task("Test"));
        tasks.add(new Task("Loop @a"));
        Collections.sort(tasks, new ContextComparator(true));
        assertEquals("Test", tasks.get(0).inFileFormat());
        assertEquals("Test @a", tasks.get(1).inFileFormat());
        assertEquals("Loop @a", tasks.get(2).inFileFormat());
        assertEquals("Test @b", tasks.get(3).inFileFormat());
    }
}

File: src/test/java/nl/mpcjanssen/simpletask/sort/ContextComparatorTest.kt
package nl.mpcjanssen.simpletask.sort

import junit.framework.TestCase
import nl.mpcjanssen.simpletask.dao.gen.TodoItem
import nl.mpcjanssen.simpletask.task.Task

import java.util.ArrayList
import java.util.Collections

class ContextComparatorTest : TestCase() {

    @Throws(Exception::class)
    fun testCompare() {
        val tasks = ArrayList<Task>()
        tasks.add(Task("Test @b"))
        tasks.add(Task("Test @a"))
        tasks.add(Task("Test"))
        tasks.add(Task("Loop @a"))
        val items = tasks.map {it -> TodoItem(0,it, false) }
        Collections.sort(items , ContextComparator(true))
        assertEquals("Test", items[0].task.inFileFormat())
        assertEquals("Test @a", items[1].task.inFileFormat())
        assertEquals("Loop @a", items[2].task.inFileFormat())
        assertEquals("Test @b", items[3].task.inFileFormat())
    }
}
